src/hotspot/os/aix/os_aix.cpp

```diff
@@ -448,6 +453,14 @@ static void query_multipage_support() {
       }
     }
 
+    // Can we use mmap with 64K pages? (Should be available with AIX7.3 TL1)
+    {
+      void* p = mmap(NULL, 1000000, PROT_READ | PROT_WRITE, MAP_ANON_64K | MAP_ANONYMOUS | MAP_SHARED, -1, 0);
+      guarantee0(p != (void*) -1); // Should always work.
```

Comment:
I probably introduced those guarantee's myself many years ago, but we actually used guarantee very rarely, only for things that absolutely have to work, or the world explodes. 

Here, there are a number of reasons why an mmap could fail, and you probably don't want to crash the VM with an guarantee at a customer site. I'd use assert.

Reply:
OK, I will use 64*K

Reply:
And what about the other guarantees in the same function after shmget() or shmat()?

Reply:
Yes, these are my faults. When I did them originally, I did not know better ( ~20 years ago). In new code, lets prefer assert.